paysage/metrics.py

```diff
@@ -361,3 +361,83 @@ def value(self) -> float:
             return (self.data_mean - self.random_mean) / math.sqrt(self.random_mean_square)
         else:
             return None
+
+class HeatCapacity(object):
+    """
+    Compute the heat capacity of the system thought of as a spin system.
+
+    We take the HC to be the second cumulant of the energy, or alternately
+    the negative second derivative with respect to inverse temperature of
+    the Gibbs free energy.
+
+    """
+
+    name = 'HeatCapacity'
+
+    def __init__(self):
+        """
+        Create HeatCapacity object.
+
+        Args:
+            downsample (int; optional): how many samples to use
+
+        Returns:
+            heat capacity object
+
+        """
+        self.heat_capacity = 0
+        self.norm = 0
+
+    def reset(self) -> None:
+        """
+        Reset the metric to it's initial state.
+
+        Note:
+            Modifies the heat capacity in place.
+
+        Args:
+            None
+
+        Returns:
+            None
+
+        """
+        self.heat_capacity = 0
+        self.norm = 0
+
+    def update(self, update_args: MetricState) -> None:
+        """
+        Update the estimate for the heat capacity
+
+        Notes:
+            Changes heat capacity in place.
+
+        Args:
+            Just the model is used since this is not a stochastic metric
+
+        Returns:
+            None
+
+        """
+        self.norm += 1
+        self.heat_capacity += be.mean(be.square(update_args.amodel
+                                   .joint_energy(update_args.random_samples)))
```

Comment:
I know we discussed this briefly, but I am not sure if this function should compute the variance using the `random_samples` or the `samples` (i.e., the fantasy particles). Have you tried it the other way?